### PR TITLE
Use configured cadl-server path in VS when using a solution

### LINKED ISSUE #660: Visual Studio extension doesn't find the cadl-server path in VSWorkspaceSettings.json when a solution is open
It seems the way we're getting to this file doesn't work unless you use "open folder" instead of "open solution". 

It seems `CurrentWorkspace` is null when using a solution: 
https://github.com/microsoft/cadl/blob/067ed4523f79b0ae40b44f546a000fc869e52cc0/packages/cadl-vs/src/VSExtension.cs#L66

It might be time to use a different file for this, but we can first check if there's an easy fix. That file has other problems: it is .gitignored by default templates and otherwise contains a bunch of temporary state that is meant to be deleted, making it a terrible place to store user settings.

cc @abatishchev

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #660

### CODE REVIEW

File: packages/cadl-vs/src/VSExtension.cs [hunk 1]
```diff
@@ -189,31 +196,36 @@ private static string GetDevelopmentCadlServerPath()
...
             var env = new Dictionary<string, string>();
             var args = new string[] { "--stdio" };
 #if DEBUG
             // Use local build of cadl-server in development (lauched from F5 in VS)
             if (InDevelopmentMode())
             {
-               var options = Environment.GetEnvironmentVariable("CADL_SERVER_NODE_OPTIONS");
-               var module = GetDevelopmentCadlServerPath();
-               return ("node.exe", new string[] { module, options }.Concat(args).ToArray(), env);
+                var options = Environment.GetEnvironmentVariable("CADL_SERVER_NODE_OPTIONS");
```
Comment (timotheeguerin) [inline packages/cadl-vs/src/VSExtension.cs:207]: hhm did the formatter not catch this before?

File: packages/cadl-vs/src/VSExtension.cs [hunk 2]
```diff
@@ -223,13 +235,59 @@ private static string GetDevelopmentCadlServerPath()
...
+            var workspace = _workspaceService?.CurrentWorkspace;
+            if (workspace != null)
+            {
+                var settings = workspace.GetSettingsManager()?.GetAggregatedSettings(SettingsTypes.Generic);
+                _configuredCadlServerPath = settings?.Property<string>("cadl.cadl-server.path");
+                _workspaceFolder = workspace.Location;
+            }
+            else
+            {
+                // When a solution is open, there is no workspace settings manager,
```
Comment (timotheeguerin) [inline packages/cadl-vs/src/VSExtension.cs:265]: Can't believe they don't have this story sorted out.

**Review by timotheeguerin — APPROVED** [review]

File: packages/cadl-vs/src/VSExtension.cs (hunk 1 above)
Comment (nguerrera) [reply packages/cadl-vs/src/VSExtension.cs:207]: It breaks with #if it seems, I ran VS format command :(

**Comment by nguerrera** [comment]
I found some bugs and made substantial changes, will open a fresh PR when ready.

**Pull request closed without merging**